table-cat/table-cat.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+"""
+Script to concatenate multiple files with common header.
+"""
+import sys
+import argparse
+import gzip
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-i", "--tables", nargs='*',
+                    help="Input multiple table with same headers")
+parser.add_argument("-z", "--gzipped", action='store_true',
+                    help="Use this option for zipped files")
+args = parser.parse_args()
+
+##################################################################
+#FIXME: Add stdin option
+# def get_filehandle(filename):
+#     """ Returns a filehandle """
+#     if filename == "-":
+#         return sys.stdin
+#     return open(filename, "r")
+#
+# ifile = get_filehandle(args.table)
+#
+# df = pd.read_csv(ifile, sep=args.idelim)
+##################################################################
+
+#Get headers
+headerlines = []
+if args.gzipped:
+    for file in args.table:
+        with gzip.open(file, 'rb') as infile:
+            first_line = infile.readline().decode('ascii')
+            headerlines += [first_line.strip()]
+else:
+    for file in args.table:
+        with open(file) as infile:
+            first_line = infile.readline()
+            headerlines += [first_line.strip()]
+
+for header1, header2 in zip(headerlines, headerlines[1:]):
```

Comment:
When does this loop end? Does this run for every combination of files? If yes how many times are files below printed to the output?

Reply:
yes, it is limited to two files only and will also create problem in double output.
Will update it!